.ci_support/build_all.py

```diff
@@ -69,13 +69,18 @@ def build_all(recipes_dir, arch):
                 text = ''.join(f.readlines())
             if platform == 'osx' and (
                     'MACOSX_DEPLOYMENT_TARGET' in text or
-                    'MACOSX_SDK_VERSION' in text):
+                    'MACOSX_SDK_VERSION' in text or
+                    'c_stdlib_version' in text):
                 config = load(text, Loader=BaseLoader)
 
                 if 'MACOSX_DEPLOYMENT_TARGET' in config:
                     for version in config['MACOSX_DEPLOYMENT_TARGET']:
                         version = tuple([int(x) for x in version.split('.')])
                         deployment_version = max(deployment_version, version)
+                if 'c_stdlib_version' in config:
+                    for version in config['c_stdlib_version']:
```

Comment:
Does this work if the values depend on the platform using selectors?

Reply:
Good question; I had assumed it would work like for `MACOSX_DEPLOYMENT_TARGET`, which can already be split between x64 and arm64, but I guess we never build arm64 in staged-recipes, so that case is not relevant.

In practice, it'll work because 10.x > 2.y even if we get some glibc versions mixed in, but I admit that that's not very nice to rely on. And I guess it would break if there's _only_  a linux `c_stdlib_version`...

Not sure how much logic I should expend here w.r.t. filtering selectors - sounds like some simple regexes would be enough; WDYT?

Reply:
So I implemented something that handles most common selectors, except ones containing `or` (like `# [linux or win]`), but those really shouldn't occur for `c_stdlib_version` / `MACOSX_DEPLOYMENT_TARGET` / `MACOSX_SDK_VERSION`, as that'd be completely wrong for at least one of the platforms.

Reply:
From my POV this works now; there's 3 specs of `c_stdlib_version` in CBC, but during the build we only take the ones for the current platform into account.

linux
```
+ echo 'Building all recipes'
Building all recipes
Found c_stdlib_version for linux: version=(2, 17)
Overriding DEFAULT_LINUX_VERSION to be cos7
```
osx

```
+ echo 'Building all recipes'
Building all recipes
Found c_stdlib_version for osx: version=(10, 12)
Overriding MACOSX_DEPLOYMENT_TARGET to be  10.12
```